source/Patches/RpcHandling.cs

```diff
@@ -172,19 +172,24 @@ private static void GenEachRole(List<GameData.PlayerInfo> infected)
                     Role.Gen<Role>(typeof(Crewmate), executioner, CustomRPC.SetExecutioner);
             }
 
-            var canHaveModifier = PlayerControl.AllPlayerControls.ToArray().ToList();
+            List<PlayerControl> canHaveModifier = PlayerControl.AllPlayerControls.ToArray().ToList().FindAll(player => player.Is(Faction.Impostors));
```

Comment:
This removing and adding from the canHaveModifier list feels very brittle and easy to mess up in the future. You've found a real bug, but I don't like this fix. I think we need to find a better method of bookkeeping.

Partially this is because the existing code is a bit messy. Probably the right answer is to do the crewmates and the impostors separately, and combine those lists. I suspect part of the key is to create separate lists and combine rather than have a single list and add to it. I dislike modifying the list in this way for the same reason that I like `final` variables; you can look at a line of code and easily run backward through the code in your head and figure out what happened to get there.